drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -243,6 +370,50 @@ local function do_configure(driver, device)
     if #wind_eps > 0 then
       profile_name = profile_name .. "-wind"
     end
+
+    if #thermo_eps > 0 then
+      profile_name = profile_name .. "-thermostat"
+      if #humidity_eps > 0 and #fan_eps > 0 then
+        profile_name = profile_name .. "-humidity" .. "-fan"
+      elseif #humidity_eps > 0 then
+        profile_name = profile_name .. "-humidity"
+      elseif #fan_eps > 0 then
+        profile_name = profile_name .. "-fan"
+      end
+    end
+
+    if #heat_eps == 0 and #cool_eps == 0 then
```

Comment:
The heat and cool eps should only be checked if thermo eps > 0, so move these under the conditional on line 374

Reply:
Move everything that is specifc to thermostat under this conditional. This includes the no-state and no-battery check